lgc/patch/ShaderInputs.cpp

```diff
@@ -343,9 +349,23 @@ void ShaderInputs::fixupUses(Module &module, PipelineState *pipelineState) {
           continue;
       }
 
-      value->setName(getInputName(static_cast<ShaderInput>(kind)));
+      auto inputKind = static_cast<ShaderInput>(kind);
+      value->setName(getInputName(inputKind));
+      const bool needCheckWorgroupId = (inputKind == ShaderInput::WorkgroupId) && (stage == ShaderStageCompute);
       for (Instruction *&call : inputUsage->users) {
         if (call && call->getFunction() == &func) {
+          if (needCheckWorgroupId) {
+            for (auto user : call->users()) {
+              if (auto extractInst = dyn_cast<ExtractElementInst>(user)) {
+                auto indexInst = dyn_cast<ConstantInt>(extractInst->getIndexOperand());
+                assert(indexInst);
```

Comment:
Please do handle the case where it isn't a constant.

Reply:
Fixed.